frontend/app/src/store/balances/blockchain-balances.ts

```diff
@@ -202,28 +243,34 @@ export const useBlockchainBalancesStore = defineStore(
       setStatus(Status.LOADED, section);
     };
 
-    const updateBlockchainTotalsState = (payload: AssetBalances) => {
-      const totals = { ...get(blockchainTotalsState), ...payload };
+    const updateBlockchainTotalsState = (
+      chain: Blockchain,
+      payload: AssetBalances
+    ) => {
+      const totals = { ...payload };
 
       for (const asset in totals) {
         if (totals[asset].amount.isZero()) delete totals[asset];
       }
 
-      set(blockchainTotalsState, totals);
+      blockchainTotalsState.value[chain] = totals;
```

Comment:
Can we keep the set approach instead of directly modifying the properties of a Ref's value?